library/ssl_msg.c

```diff
@@ -2048,7 +2048,7 @@ int mbedtls_ssl_fetch_input( mbedtls_ssl_context *ssl, size_t nb_want )
             if( ret < 0 )
                 return( ret );
 
-            if ( (size_t)ret > len || ( INT_MAX > SIZE_MAX && ret > SIZE_MAX ) )
+            if ( (size_t)ret > len || ( INT_MAX > SIZE_MAX && (size_t)ret > SIZE_MAX ) )
```

Comment:
[I think we should actually remove the handling of `INT_MAX > SIZE_MAX`](https://github.com/ARMmbed/mbedtls/issues/3577) (which was added in https://github.com/ARMmbed/mbedtls/pull/1395#discussion_r177467078). We don't actually support platforms where `SIZE_MAX` is smaller than `INT_MAX` because when we review the code, we assume that a value of type `size_t` will not be promoted. But this is out of scope here.

Reply:
Should I then eliminate this check from here? Should I add this check to `CMakeLists.txt` also? Or may it require deeper investigations and it's better not touch this now?

Reply:
Please leave things as they are. I filed an issue; let's give time for people to react if they think it's a bad idea.

Reply:
Okay.